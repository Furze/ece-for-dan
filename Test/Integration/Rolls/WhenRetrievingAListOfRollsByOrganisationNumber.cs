using System.Linq;
using MoE.ECE.CLI.Data;
using MoE.ECE.Domain.Read.Model;
using MoE.ECE.Integration.Tests.Chapter;
using MoE.ECE.Integration.Tests.Infrastructure;
using Shouldly;
using Xunit;
using Xunit.Abstractions;

namespace MoE.ECE.Integration.Tests.Rolls
{
    public class WhenRetrievingAListOfRollsByOrganisationNumber : SpeedyIntegrationTestBase
    {
        private const string Rs7 = "Rs7";
        private readonly int? _refOrganisationId = ReferenceData.EceServices.MontessoriLittleHands.RefOrganisationId;

        public WhenRetrievingAListOfRollsByOrganisationNumber(RunOnceBeforeAllTests testSetUp, ITestOutputHelper output,
            TestState<ECEStoryBook, ECEStoryData> testState) : base(testSetUp, output, testState)
        {
        }


        protected override void Arrange() =>
            Given
                .A_rs7_skeleton_has_been_created(submission =>
                {
                    submission.OrganisationId = ReferenceData.EceServices.MontessoriLittleHands.RefOrganisationId;
                    submission.FundingPeriodYear = 2020;
                })
                .The_rs7_has_been_submitted_for_peer_approval();

        protected override void Act() => When.Get($"api/rolls?organisation-id={_refOrganisationId}");

        [Fact]
        public void ThenTheResponseShouldBeOk() => Then.Response.ShouldBe.Ok();

        [Fact]
        public void ThenTheResponseShouldBeOfTheCorrectType() =>
            Then.Response.ShouldBe.Ok<CollectionModel<RollModel>>();

        [Fact]
        public void Then_the_response_snapshot_should_be_ok() =>
            Then.Snapshot().Match<CollectionModel<RollModel>>(IgnoreFieldsFor.CollectionModelRollModel);

        [Fact]
        public void ThenTheResponseShouldContainAnRs7Roll()
        {
            var response = Then.Response.Content<CollectionModel<RollModel>>();

            response.Data.ShouldContain(model => model.RollType == Rs7);
        }

        [Fact]
        public void ThenTheRs7RollShouldHaveAPopulatedBusinessEntityId()
        {
            var response = Then.Response.Content<CollectionModel<RollModel>>();

            response.Data.First(model => model.RollType == Rs7).BusinessEntityId.ShouldNotBeNull();
        }

        [Fact]
        public void ThenTheRs7RollShouldHaveAPopulatedStatus()
        {
            var response = Then.Response.Content<CollectionModel<RollModel>>();

            response.Data.First(model => model.RollType == Rs7).Status.ShouldNotBeNull();
        }
    }
}