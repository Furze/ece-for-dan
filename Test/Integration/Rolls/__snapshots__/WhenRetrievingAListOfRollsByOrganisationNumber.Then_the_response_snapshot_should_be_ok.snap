{
  "Data": [
    {
      "Id": 114001,
      "BusinessEntityId": "28a9f0f6-9bfd-4fc0-81e2-01e82d6a4dc1",
      "OrganisationId": 114895,
      "RollType": "Rs7",
      "Received": "2020-11-30T15:38:49.575458+13:00",
      "Status": "ExternalSubmittedForApproval",
      "AssignedTo": null,
      "FundingPeriodMonth": "July",
      "FundingYear": 2021,
      "FundingPeriodYear": 2020
    }
  ],
  "Pagination": {
    "PageSize": 10,
    "PageNumber": 1,
    "Count": 1
  }
}
